### PR TITLE
Improve to can specify parameters on issue comments api.

### PR DESCRIPTION
**Pull request by t-kuni**
I had try to get comments which updated after specific date but it couldn't because can't specify `since` parameters through this library.

Github API provide `since` parameter.

https://developer.github.com/v3/issues/comments/#list-comments-on-an-issue
image

but this library can specify `page` only.
https://github.com/KnpLabs/php-github-api/blob/master/lib/Github/Api/Issue/Comments.php#L39-L56

In this PR, will be able to specify array on 4th argument of Comments#all method.

### CODE REVIEW

File: lib/Github/Api/Issue/Comments.php [hunk 1]
```diff
@@ -41,18 +41,24 @@ public function configure($bodyType = null)
...
      */
     public function all($username, $repository, $issue, $page = 1)
     {
-        return $this->get('/repos/'.rawurlencode($username).'/'.rawurlencode($repository).'/issues/'.rawurlencode($issue).'/comments', [
-            'page' => $page,
-        ]);
+        if (is_array($page)) {
+            $parameters = $page;
+        } else {
+            $parameters = [
```
Comment (acrobat) [inline lib/Github/Api/Issue/Comments.php]: Can you add a deprecation warning here, indicating that it's deprecated to pass a different type other than array to the `$page` parameter and that an array will be required in the future?

`@trigger_error('message', E_USER_DEPRECATED);`

File: lib/Github/Api/Issue/Comments.php [hunk 2]
```diff
@@ -41,18 +41,24 @@ public function configure($bodyType = null)
...
      * @link https://developer.github.com/v3/issues/comments/#list-comments-on-an-issue
      *
-     * @param string $username
-     * @param string $repository
-     * @param int    $issue
-     * @param int    $page
+     * @param string    $username
+     * @param string    $repository
+     * @param int       $issue
+     * @param int|array $page
```
Comment (acrobat) [inline lib/Github/Api/Issue/Comments.php]: Also add a description to the page parameter, to indicate that it's deprecated to provide an integer here

File: lib/Github/Api/Issue/Comments.php (hunk 1 above)
Comment (t-kuni) [reply lib/Github/Api/Issue/Comments.php]: Thank you for your reply.
I added a @trigger_error and a comment with reference to the following code.
https://github.com/KnpLabs/php-github-api/blob/6e843e62160bb586ece922026226ea6a00fc1f74/lib/Github/Api/Issue/Assignees.php#L61

**Comment by acrobat** [comment]
Thanks @t-kuni! And congrats on your first contribution! 🎉

**Comment by bobeagan** [comment]
How is this supposed to play nice with ResultPager's fetchAll on the comments->all() method?

**Pull request merged**